src/FSharp.Core/async.fs

```diff
@@ -868,11 +868,7 @@ module AsyncPrimitives =
 
     ///   - Initial cancellation check
     ///   - Call syncCtxt.Post with exception protection. This may fail as it is arbitrary user code
-#if BUILDING_WITH_LKG || NO_NULLCHECKING_LIB_SUPPORT
```

Comment:
Thank you for this contribution.
This reminds me that with .NET9 release, we do not need most (maybe all?) of these  directives anymore - those were mainly needed when the old compiler used for building this compiler did not understand the new syntax.

There might still be situations for nullness-related bugs where we would want to build the compiler with nullness checking disabled (e.g. if there is a regression not allowing it any other way), but the syntax should be just fine.